alicloud/resource_alicloud_launch_template.go

```diff
@@ -0,0 +1,490 @@
+	d.Set("internet_max_bandwidth_out", template.LaunchTemplateData.InternetMaxBandwidthOut)
+	d.Set("io_optimized", template.LaunchTemplateData.IoOptimized)
+	d.Set("key_pair_name", template.LaunchTemplateData.KeyPairName)
+	d.Set("network_type", template.LaunchTemplateData.NetworkType)
+	d.Set("ram_role_name", template.LaunchTemplateData.RamRoleName)
+	d.Set("resource_group_id", template.LaunchTemplateData.ResourceGroupId)
+	d.Set("security_enhancement_strategy", template.LaunchTemplateData.SecurityEnhancementStrategy)
+	d.Set("security_group_id", template.LaunchTemplateData.SecurityGroupId)
+	d.Set("spot_price_limit", template.LaunchTemplateData.SpotPriceLimit)
+	d.Set("spot_strategy", template.LaunchTemplateData.SpotStrategy)
+	d.Set("system_disk_name", template.LaunchTemplateData.SystemDiskDiskName)
+	d.Set("system_disk_category", template.LaunchTemplateData.SystemDiskCategory)
+	d.Set("system_disk_description", template.LaunchTemplateData.SystemDiskDescription)
+	d.Set("system_disk_size", template.LaunchTemplateData.SystemDiskSize)
+	d.Set("resource_group_id", template.LaunchTemplateData.ResourceGroupId)
+	d.Set("userdata", template.LaunchTemplateData.UserData)
+	d.Set("vswitch_id", template.LaunchTemplateData.VSwitchId)
+	d.Set("vpc_id", template.LaunchTemplateData.VpcId)
+	d.Set("zone_id", template.LaunchTemplateData.ZoneId)
+
+	var interfaces []map[string]interface{}
+	for _, net := range template.LaunchTemplateData.NetworkInterfaces.NetworkInterface {
+		ds := make(map[string]interface{})
+		ds["vswitch_id"] = net.VSwitchId
+		ds["security_group_id"] = net.SecurityGroupId
+		ds["name"] = net.NetworkInterfaceName
+		ds["description"] = net.Description
+		ds["primary_ip"] = net.PrimaryIpAddress
+		interfaces = append(interfaces, ds)
+	}
+	if err := d.Set("network_interfaces", interfaces); err != nil {
+		return err
+	}
+
+	var disks []map[string]interface{}
+	for _, disk := range template.LaunchTemplateData.DataDisks.DataDisk {
+		ds := make(map[string]interface{})
+		ds["size"] = disk.Size
+		ds["snapshot_id"] = disk.SnapshotId
+		ds["category"] = disk.Category
+		ds["encrypted"] = (disk.Encrypted == "true")
+		ds["name"] = disk.DiskName
+		ds["description"] = disk.Description
+		ds["delete_with_instance"] = disk.DeleteWithInstance
+		disks = append(disks, ds)
+	}
+	if err := d.Set("data_disks", disks); err != nil {
+		return err
+	}
+
+	tags := make(map[string]interface{})
+	for _, tag := range template.LaunchTemplateData.Tags.InstanceTag {
+		tags[tag.Key] = tag.Value
+	}
+	d.Set("tags", tags)
+
+	return nil
+}
+
+func resourceAliyunLaunchTemplateDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+
+	req := ecs.CreateDeleteLaunchTemplateRequest()
+	req.LaunchTemplateId = d.Id()
+
+	_, err := client.WithEcsClient(func(ecsClient *ecs.Client) (interface{}, error) {
+		return ecsClient.DeleteLaunchTemplate(req)
+	})
+	if err != nil {
+		if NotFoundError(err) {
```

Comment:
DeleteLaunchTemplate's error is not a `NotFoundError `, please use `IsExpectedErrors` instead.